### PR TITLE
Allow migrations from integer to string without size limit

### PR DESCRIPTION
**Pull request by cliedeman**

### CODE REVIEW

File: entc/integration/migrate/entv1/schema/user.go [hunk 1]
```diff
@@ -41,6 +41,8 @@ func (User) Fields() []ent.Field {
 		field.String("workplace").
 			MaxLen(30).
 			Optional(),
+		field.Int64("int64_to_string").Optional(),
+		field.Int64("uint64_to_string").Positive().Optional(),
```
Comment (a8m) [inline entc/integration/migrate/entv1/schema/user.go:45]:
```suggestion
		field.Int64("int64_to_string").
			Optional(),
		field.Uint64("uint64_to_string").
			Positive().
			Optional(),
```

File: entc/integration/migrate/entv2/schema/user.go [hunk 2]
```diff
@@ -80,6 +80,10 @@ func (User) Fields() []ent.Field {
 		field.String("workplace").
 			Optional(),
 		// deleting the `address` column.
+		// Change int64 to string
+		field.String("int64_to_string").Optional(),
+		// Change uint64 to string
+		field.String("uint64_to_string").Optional(),
```
Comment (a8m) [inline entc/integration/migrate/entv2/schema/user.go:86]:
```suggestion
		// Change int64 to string.
		field.String("int64_to_string").
			Optional(),
		// Change uint64 to string.
		field.String("uint64_to_string").
			Optional(),
```

File: entc/integration/migrate/migrate_test.go [hunk 3]
```diff
@@ -56,6 +56,8 @@ func TestPostgres(t *testing.T) {
 			require.NoError(t, err)
 			defer root.Close()
 			ctx := context.Background()
+			err = root.Exec(ctx, "DROP DATABASE IF EXISTS migrate", []interface{}{}, new(sql.Result))
+			require.NoError(t, err)
```
Comment (a8m) [inline entc/integration/migrate/migrate_test.go:60]: 👍

**Review by a8m — APPROVED** [review]
Great work @cliedeman! Thanks for addressing this.

**Comment by cliedeman** [comment]
@a8m Thanks this is good to go.

I started on the next step and encountered some issues

I created a new table dedicated to conversions

```
		// Conversions
		client.Conversion.Create().
			SetName("zero").
			SetInt8ToString(0).
			SetUint8ToString(0).
			SetInt16ToString(0).
			SetUint16ToString(0).
			SetInt32ToString(0).
			SetUint32ToString(0).
			SetInt64ToString(0).
			SetUint64ToString(0).
			SaveX(ctx)

		client.Conversion.Create().
			SetName("min").
			SetInt8ToString(math.MinInt8).
			SetUint8ToString(0).
			SetInt16ToString(math.MinInt16).
			SetUint16ToString(0).
			SetInt32ToString(math.MinInt32).
			SetUint32ToString(0).
			SetInt64ToString(math.MinInt64).
			SetUint64ToString(0).
			SaveX(ctx)

		client.Conversion.Create().
			SetName("max").
			SetInt8ToString(math.MaxInt8).
                          // TODO: postgres does not support uint types
			// SetUint8ToString(math.MaxUint8).
			SetUint8ToString(math.MaxInt8).
			SetInt16ToString(math.MaxInt16).
			// SetUint16ToString(math.MaxUint16).
			SetUint16ToString(math.MaxInt16).
			SetInt32ToString(math.MaxInt32).
			// SetUint32ToString(math.MaxUint32).
			SetUint32ToString(math.MaxInt32).
			SetInt64ToString(math.MaxInt64).
			// https://github.com/golang/go/issues/20238 :(
			SetUint64ToString(math.MaxInt64).
			SaveX(ctx)
```

but this caused the idRange tests to fail.

I think I should create a dedicated project (`entc/migrate/conversions`) for these tests

Other Notes:
 * https://github.com/golang/go/issues/20238
 * Postgres does not support unit types at all. So I need to adjust the tests depending on the db

**Comment by a8m** [comment]
> but this caused the idRange tests to fail.

You need to update the id-range tests as well.

**Comment by a8m** [comment]
Thanks for addressing this issue @cliedeman, you're welcome to add the conversion tests in another PR.

**Pull request merged**